src/Employer.Web/Orchestrators/TrainingProviderOrchestrator.cs

```diff
@@ -20,22 +23,53 @@ public TrainingProviderOrchestrator(IVacancyClient client)
 
             var vm = new IndexViewModel
             {
-                Title = vacancy.Title
+                Title = vacancy.Title,
+                Ukprn = vacancy.Ukprn,
+                ProviderName = vacancy.ProviderName,
+                ProviderAddress = vacancy.ProviderAddress
             };
 
             return vm;
         }
 
-        public async Task<ConfirmViewModel> GetConfirmViewModelAsync(Guid vacancyId)
+        public async Task<ConfirmViewModel> PostIndexEditModelAsync(IndexEditModel m)
         {
-            var vacancy = await _client.GetVacancyForEditAsync(vacancyId);
-
-            var vm = new ConfirmViewModel
+            var vacancy = await _client.GetVacancyForEditAsync(m.VacancyId);
+            
+            var confirmViewModel = new ConfirmViewModel
             {
                 Title = vacancy.Title
             };
 
-            return vm;
+            if (long.TryParse(m.Ukprn, out var ukprn) && ukprn != vacancy.Ukprn)
+            {
+                var providerDetail = await _providerService.GetProviderDetailAsync(ukprn);
+                confirmViewModel.Ukprn = providerDetail.Ukprn;
+                confirmViewModel.ProviderName = providerDetail.ProviderName;
+                confirmViewModel.ProviderAddress = providerDetail.ProviderAddress;
+            }
+            else
+            {
+                confirmViewModel.Ukprn = vacancy.Ukprn.Value;
+                confirmViewModel.ProviderName = vacancy.ProviderName;
+                confirmViewModel.ProviderAddress = vacancy.ProviderAddress;
+            }
+
+            return confirmViewModel;
+        }
+
+        public async Task PostConfirmEditModelAsync(ConfirmEditModel m)
+        {
+            var vacancy = await _client.GetVacancyForEditAsync(m.VacancyId);
+            vacancy.Ukprn = long.Parse(m.Ukprn);
+            vacancy.ProviderName = m.ProviderName;
+            vacancy.ProviderAddress = m.ProviderAddress;
+            await _client.UpdateVacancyAsync(vacancy, canUpdateQueryStore: false);
```

Comment:
For security reasons I wouldn't assume Ukprn, ProviderName and Address are correct at this point. I'd just post back the Ukprn in the model and re-validate it and then get name and address from the provider service.